network/network.go

```diff
@@ -510,14 +510,20 @@ func (n *Network) BroadcastByAddresses(message proto.Message, addresses ...strin
 
 // BroadcastByIDs broadcasts a message to a set of peer clients denoted by their peer IDs.
 func (n *Network) BroadcastByIDs(message proto.Message, ids ...peer.ID) {
-	signed, err := n.PrepareMessage(message)
-	if err != nil {
-		return
-	}
+	n.Peers.Range(func(key, value interface{}) bool {
+		client := value.(*PeerClient)
 
-	for _, id := range ids {
-		n.Write(id.Address, signed)
-	}
+		if !isIn(client.Address, ids...) {
```

Comment:
ids should be smaller than n.Peers. Feels strange that we're looping through all the peers and matching for the id rather than just loop through the ids given. Any reason for the switch?

Reply:
Good catch. Haven't thought about sizes of `ids` & `n.Peers`